### DISCUSSION TITLE
Server crash, start limit hit

### DISCUSSION THREAD

**Discussion by Ollie-Gutierrez** [Infinidoge/nix-minecraft#118] [Q&A] [answered]
Hi, my server seems to be crashing but i dont really know why, from the errors it seems like c2me, but ive tried running the mods and configs from just a normal server jar, and the server works fine. I've added my nixos configurations, and the latest log below:

minecraft.nix:
```
{
  inputs,
  pkgs,
  ...
}: let
  modpack = pkgs.fetchzip {
    url = "https://0x0.st/8sQH.1.zip";
    hash = "sha256-zO6OAkOnUkvPqJXW7ffn9b3pjlzb5MAKQJVJK574Un0=";
    stripRoot = false;
  };
in {
  imports = [inputs.nix-minecraft.nixosModules.minecraft-servers];
  nixpkgs.overlays = [inputs.nix-minecraft.overlay];
  environment.systemPackages = with pkgs; [ngrok];
  services.minecraft-servers = {
    enable = true;
    eula = true;
    openFirewall = true;
    servers.valhalla = {
      enable = true;
      package = pkgs.fabricServers.fabric-1_20_1;
      serverProperties = {
        server-ip = "0.0.0.0";
        server-port = 25566;
      };

      symlinks = {
	"mods" = "${modpack}/minecraft/mods";
	"config" = "${modpack}/minecraft/config";
      };
    };
  };
  systemd.services.cloudflared-minecraft = {
    description = "Minecraft cloudflared Tunnel";
    after = ["network.target"];
    serviceConfig = {
      ExecStart = "${pkgs.cloudflared}/bin/cloudflared --config /etc/cloudflared/minecraft-config.yml tunnel run";
      Restart = "always";
      RestartSec = "10s";
      User = "root"; # You can set to a non-privileged user if desired
    };

    wantedBy = ["multi-user.target"];
  };

  networking.firewall.allowedTCPPorts = [25566];
}
```
[latest.log](https://github.com/user-attachments/files/18267259/latest.log)

**Comment by Infinidoge** [comment, marked as answer]
The issue appears to be that Very Many Players (vmp) is trying to write to its config file, causing it to crash due to trying to write to a read-only filesystem.

**Reply by Ollie-Gutierrez** [reply]
Thanks for you're help, so i should be using the servers.<name>.files option then for configs? How do i use this exactly, i tried using it in the same way as symlink but it didn't work.

**Reply by Infinidoge** [reply]
Yes, files is how you would fix this. If you update `nix-minecraft` it should now work, as #116 made `files` copy directories. Note however that this will overwrite the files each time to keep them the same, so if you change something in game (or in this case, let it be changed by the mod), the changes will be erased the next time it restarts.
